fastlane_core/lib/fastlane_core/configuration/config_item.rb

```diff
@@ -240,6 +241,12 @@ def auto_convert_value(value)
       elsif allow_shell_conversion
         return value.shelljoin if value.kind_of?(Array)
         return value.map { |k, v| "#{k.to_s.shellescape}=#{v.shellescape}" }.join(' ') if value.kind_of?(Hash)
+      elsif data_type == Hash && value.kind_of?(String)
+        begin
+          parsed = JSON.parse(value)
+          return parsed if parsed.kind_of?(Hash)
+        rescue JSON::ParserError
+        end
```

Comment:
I would make sure we return something in the case of a `ParserError`

`retun value` seems like a non-breaking change.

Reply:
Hmm. I wanted to preserve the existing behavior, which basically throws an error saying expecting Hash but got String

I could let the JSON::ParserError surface up if that's preferable?

Reply:
Ah yes! It does `return value` at the end of the function, so it's effectively the same. This looks good.